### PR TITLE
lago-550 bdus xusd pool

### PR DESCRIPTION
**Pull request by AndrzejGisWideLab**
## What's the purpose of your PR?

- [ ] Fixing a bug
- [x] A new Feature
- [ ] Techdebt
- [ ] General Task

### CODE REVIEW

**Comment by AndrzejGisWideLab** [comment]
don't merge until we run deployment scripts on prod

File: deploy/9_add_bdus_xusd_pool.ts
Comment (AndrzejGisWideLab) [inline deploy/9_add_bdus_xusd_pool.ts:10]: I added this guard so it warns us in case we deploy to another network

File: utils/Constants.ts
Comment (AndrzejGisWideLab) [inline utils/Constants.ts:90]: I had to find XUSD replacement on local fork. I chose usdc for no particular reason

File: utils/SystemSetup.ts
Comment (AndrzejGisWideLab) [inline utils/SystemSetup.ts:117]: I wasn't sure if we wanted to add some generic, mandatory support for other chains in the future, so I decided to go with YAGNI and added it only to local deployment to test FE. WDYT?

File: utils/abis/mint-abi.json
Comment (AndrzejGisWideLab) [inline utils/abis/mint-abi.json:1]: I didn't use it after all, but learned that this can be used instead of contract name when you call `hre.ethers.getContractAt(abiJsonObjectFromThisJson, address)` you can provide just a small part of the abi and call the function you need to call. Pretty useful for external contract communication. I decided to keep this abi, even though it's not used anywhere. WDYT?

File: deploy/9_add_bdus_xusd_pool.ts
Comment (tkeidar) [inline deploy/9_add_bdus_xusd_pool.ts:16]: I like the name externalUsdStable more than supplementary in this case. maybe we should change it also on `constants`?

File: utils/LocalHelpers.ts
Comment (tkeidar) [inline utils/LocalHelpers.ts:6]: I don't get this function what do we use it for?
both its name and usage are not clear

File: utils/DeployedContractsHelpers.ts
Comment (tkeidar) [inline utils/DeployedContractsHelpers.ts:382]:
```suggestion
  } else if (contractName === SUPPLEMENTARY_USD_STABLE_NAME["rsk"]) {
```

Comment (tkeidar) [reply utils/DeployedContractsHelpers.ts:382]: you can either do that for both rsk and mainnetFork or going with this generic approach:

```
else if (contractName === SUPPLEMENTARY_USD_STABLE_NAME[hre.network.name]) {
    return SUPPLEMENTARY_USD_STABLE_ADDRESS[hre.network.name]
}
```

File: utils/UniswapPoolsHelpers.ts
Comment (tkeidar) [inline utils/UniswapPoolsHelpers.ts:121]: we will sure `registerPool()` also for RSK and other networks, so we need this variable (usdc) to be generic

File: utils/SystemSetup.ts
Comment (tkeidar) [reply utils/SystemSetup.ts:117]: I don't think it's something we want to deal with, atm

File: utils/abis/mint-abi.json
Comment (tkeidar) [reply utils/abis/mint-abi.json:1]: cool, it can be very useful - so great to know that 😎
I wonder whether a future version of us (or another developer) will understand why this file is here. I don't think we should keep it

File: utils/Constants.ts
Comment (kazazor) [reply utils/Constants.ts:110]: And actually, let's go with the approach we did in the other tokens in this fle. WDYT? keeps things a bit more standard:
```suggestion
export const SUPPLEMENTARY_USD_STABLE: { [key: string]: {symbol:string, address:string} } = {
  mainnetFork: {symbol: "USDC", address: ETH_USDC_ADDRESS},
  rsk: {symbol: "XUSD", address: RSK_XUSD_ADDRESS}
};
```

**Review by kazazor — APPROVED** [review]
I added my comments, looks very nice! I do agree with Tom's comments so I'm approving it and letting you continue the discussion on those.

Please note that both eslint & the tests are failing, so you won't be able to merge until fixed.

File: utils/DeployedContractsHelpers.ts
Comment (tkeidar) [inline utils/DeployedContractsHelpers.ts:388]: line 110 `getAllBDStableStakingRewards()` - should add also the new staking pool

**Review by tkeidar — APPROVED** [review]
Looks good 👍 
@AndrzejGisWideLab Approving this PR although I had some comments, please go over it and let me know if you think I should take another look before merging

File: utils/abis/mint-abi.json
Comment (AndrzejGisWideLab) [reply utils/abis/mint-abi.json:1]: You're right, I got too excited about it ;)

File: utils/DeployedContractsHelpers.ts
Comment (AndrzejGisWideLab) [reply utils/DeployedContractsHelpers.ts:382]: very good point, thanks

File: utils/Constants.ts
Comment (AndrzejGisWideLab) [reply utils/Constants.ts:110]: it's actually very neat, thanks

File: utils/DeploymentHelpers.ts
Comment (tkeidar) [inline utils/DeploymentHelpers.ts:61]:
```suggestion
  console.log("registered staking rewards pool to staking rewards distribution");
```

**Review by tkeidar — APPROVED** [review]

**Pull request merged**
